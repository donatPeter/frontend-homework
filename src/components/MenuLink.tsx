import styled from "styled-components";
import { Icon } from "./Icon";

type Props = {
  text: string;
  icon: string;
  href: string;
  color?: string;
};

export const StyledLink = styled.a<{ $color: string }>`
  display: flex;
  text-decoration: none;
  color: ${({ $color }) => $color};
  align-items: center;
  padding: 6px 16px;
`;

export const MenuText = styled.span`
  margin-left: 8px;
`;

const MenuLink = ({ icon, text, href, color }: Props) => (
  <StyledLink href={href} $color={color || "inherit"}>
    <Icon icon={icon} size="20px" />
    <MenuText>{text}</MenuText>
  </StyledLink>
);

export default MenuLink;
