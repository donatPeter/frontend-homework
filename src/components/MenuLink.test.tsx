import { shallow } from "enzyme";
import MenuLink from "./MenuLink";

describe("<MenuLink />", () => {
  it("renders link", () => {
    expect(
      shallow(<MenuLink href="" icon="phone" text="Call me" />)
    ).toMatchSnapshot();
  });
});
