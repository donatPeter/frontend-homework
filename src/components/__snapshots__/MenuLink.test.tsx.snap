// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> renders link 1`] = `
<styled.a
  $color="inherit"
  href=""
>
  <Icon
    icon="phone"
    size="20px"
  />
  <styled.span>
    Call me
  </styled.span>
</styled.a>
`;
